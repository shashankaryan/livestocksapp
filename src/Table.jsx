// @flow

import React, { Component } from 'react'
import TableItem from './TableItem'

class Table extends Component {
	state = {}
	props: {
		items: Array<Item>,
	}

	render() {
		return (
			<div>
				<table className="container">
					<thead>
						<tr>
							<th>
								<h1>Ticker</h1>
							</th>
							<th>
								<h1>Price</h1>
							</th>
							<th>
								<h1>Last Update</h1>
							</th>
						</tr>
					</thead>
					<tbody>{this.props.items.map(item => <TableItem key={item.name} {...item} />)}</tbody>
				</table>
			</div>
		)
	}
}
export default Table
