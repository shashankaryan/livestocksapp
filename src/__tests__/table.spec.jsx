import React from 'react'
import { shallow } from 'enzyme' // render, static is also available. but they are slower.
import Table from '../Table'

test('Table should render correctly based on items', () => {
	const items = [{}, {}, {}, {}, {}]
	const component = shallow(<Table items={items} />)
	expect(component).toMatchSnapshot()
})
