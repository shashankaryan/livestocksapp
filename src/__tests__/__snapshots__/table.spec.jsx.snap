// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly based on items 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <table
        className="container"
    >
        <thead>
            <tr>
                <th>
                    <h1>
                        Ticker
                    </h1>
                </th>
                <th>
                    <h1>
                        Price
                    </h1>
                </th>
                <th>
                    <h1>
                        Last Update
                    </h1>
                </th>
            </tr>
        </thead>
        <tbody>
            <TableItem />
            <TableItem />
            <TableItem />
            <TableItem />
            <TableItem />
        </tbody>
    </table>
</div>,
  "nodes": Array [
    <div>
      <table
            className="container"
      >
            <thead>
                  <tr>
                        <th>
                              <h1>
                                    Ticker
                              </h1>
                        </th>
                        <th>
                              <h1>
                                    Price
                              </h1>
                        </th>
                        <th>
                              <h1>
                                    Last Update
                              </h1>
                        </th>
                  </tr>
            </thead>
            <tbody>
                  <TableItem />
                  <TableItem />
                  <TableItem />
                  <TableItem />
                  <TableItem />
            </tbody>
      </table>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Table
        items={
                Array [
                        Object {},
                        Object {},
                        Object {},
                        Object {},
                        Object {},
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Table {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "items": Array [
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
          ],
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <table
                    className="container"
          >
                    <thead>
                              <tr>
                                        <th>
                                                  <h1>
                                                            Ticker
                                                  </h1>
                                        </th>
                                        <th>
                                                  <h1>
                                                            Price
                                                  </h1>
                                        </th>
                                        <th>
                                                  <h1>
                                                            Last Update
                                                  </h1>
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <TableItem />
                              <TableItem />
                              <TableItem />
                              <TableItem />
                              <TableItem />
                    </tbody>
          </table>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <table
                    className="container"
          >
                    <thead>
                              <tr>
                                        <th>
                                                  <h1>
                                                            Ticker
                                                  </h1>
                                        </th>
                                        <th>
                                                  <h1>
                                                            Price
                                                  </h1>
                                        </th>
                                        <th>
                                                  <h1>
                                                            Last Update
                                                  </h1>
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <TableItem />
                              <TableItem />
                              <TableItem />
                              <TableItem />
                              <TableItem />
                    </tbody>
          </table>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Table
    items={
        Array [
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
          ]
    }
/>,
}
`;
